src/librustc_resolve/late.rs

```diff
@@ -1157,6 +1175,9 @@ impl<'a, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
         impl_items: &'ast [P<AssocItem>],
     ) {
         debug!("resolve_implementation");
+        let old_ignore = self.in_func_body;
+        // Never ignore errors in trait implementations.
+        self.in_func_body = false;
```

Comment:
```suggestion
        let old_ignore = mem::replace(&mut this.in_func_body, false);
```
Although I'm not sure why this special case exists.

Reply:
> I'm not sure why this special case exists.

Trait implementations are publicly visible outside the current scope. See https://github.com/rust-lang/rust/pull/73566/files#diff-f2b1241f9e58664507f0cc3a865bfb87R24 for an example - rustdoc can't document that trait impl unless name resolution succeeded.

Reply:
@jyn514 I would expect this to set `in_func_body` to `false` for the duration of `visit_item` (and reset it back to the previous value afterwards).

Reply:
Isn't that the current behavior? Or do you mean it should happen for _every_ item, not just trait implementations?